### PR TITLE
Remove queue from the Prometheus remote write

### LINKED ISSUE #2949: Queued Retry is incompatible with Prometheus Remote Write
Prometheus remote write (PWR) exporter is supporting queued retry out of the box, but due to the strict requirements of PWR, the queue causes "out of order sample" errors. For PWR to work correctly, all the samples sent from the collector need to be in chronological order for a time series. If sending queue is enabled:

```
sending_queue:
   enabled: true
   num_consumers: 5 
```

It causes the samples from the same timeseries to be consumed by different queued and causes the ordering issue I experience.

cc @odeke-em

### PR DESCRIPTION
**Pull request by rakyll**
Prometheus remote write (PWR) exporter is supporting queued retry
out of the box, but due to the strict requirements of PWR, the queue
causes "out of order sample" errors. The queue won't be suitable
for Prometheus unless it has a way to shard the data by timeseries.

This is not going to break the existing users because if they have
this feature enabled, they already can't use the PRW exporter correctly.

Fixes #2949.

### CODE REVIEW

**Comment by rakyll** [comment]
cc @odeke-em

File: exporter/prometheusremotewriteexporter/README.md [hunk 1]
```diff
@@ -51,4 +51,4 @@ Several helper files are leveraged to provide additional capabilities automatica
 
 - [HTTP settings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/README.md)
 - [TLS and mTLS settings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md)
-- [Queuing, retry and timeout settings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md)
+- [Retry and timeout settings](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/README.md), not that the exporter doesn't support `sending_queue`.
```
Comment (odeke-em) [inline exporter/prometheusremotewriteexporter/README.md:54]: This is a double negative, could we perhaps make it:

    The exporter does not support sending_queue

**Review by odeke-em — APPROVED** [review]
LGTM, thank you @rakyll! I added a suggestion but also would be nice to add the issue you filed, as a reference to the code comments. Thank you.

File: exporter/prometheusremotewriteexporter/README.md (hunk 1 above)
Comment (rakyll) [reply exporter/prometheusremotewriteexporter/README.md:54]: Sorry, it was a typo. I wanted to say "note that...". Fixed now.

**Comment by odeke-em** [comment]
Kindly cc-ing @bogdandrutu!

**Comment by rakyll** [comment]
Added a link to the issue, thanks for the suggestion @odeke-em.

**Comment by odeke-em** [comment]
Nice, thank you @rakyll! LGTM.

**Comment by tigrannajaryan** [comment]
Can we instead force queued_retry to have a single consumer for this exporter? That should fix the problem while keeping the retrying and queuing capabilities, right? I am not sure how important is that for this exporter though.

**Comment by rakyll** [comment]
@tigrannajaryan Retrying is critical in the context of remote write, it's not easy to have a highly available Prometheus, Cortex or Thanos. But we want to reimplement the retrying options Prometheus server implements natively (https://prometheus.io/docs/practices/remote_write/) to provide a drop-in replacement.

**Comment by tigrannajaryan** [comment]
> @tigrannajaryan Retrying is critical in the context of remote write, it's not easy to have a highly available Prometheus, Cortex or Thanos. We want to reimplement the retrying options Prometheus server implements natively (https://prometheus.io/docs/practices/remote_write/) to provide a drop-in replacement and help users to reuse their fine tuned configuration.

Sounds good. Do you also want to add a TODO or an issue (if it does not already exist) describing what you plan to do?

**Review by tigrannajaryan — APPROVED** [review]

**Comment by rakyll** [comment]
I updated https://github.com/open-telemetry/opentelemetry-collector/issues/2259 to make sure we're following up with the necessary changes. Thanks much.

**Pull request merged**